### PR TITLE
Implemented NATS preconfigured module.

### PR DESCRIPTION
**Pull request by gmamaladze**

### CODE REVIEW

File: src/modules/nats/nats-container.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const SUBJECT = "HELLO";
+    const PAYLOAD = "WORLD";
+
+    const container = await new NatsContainer().start();
+    const nc = await connect(container.getConnectionOptions());
+    const sc = StringCodec();
+
+    //----------------
+    const sub = nc.subscribe(SUBJECT);
+    (async () => {
```
Comment (cristianrgreco) [inline src/modules/nats/nats-container.test.ts:31]: Is the IIFE required/can the console.log be removed?

File: src/modules/nats/nats-container.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { GenericContainer } from "../../generic-container/generic-container";
+import { StartedTestContainer } from "../../test-container";
+import { RandomUuid } from "../../uuid";
+import { AbstractStartedContainer } from "../abstract-started-container";
+import { ConnectionOptions } from "nats/lib/nats-base-client/types";
```
Comment (cristianrgreco) [inline src/modules/nats/nats-container.ts:5]: You should create your own type here, as we cannot include the `nats` library as a dependency

File: src/modules/nats/nats-container.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        result.push(args[argsKey]);
+      }
+      return result;
+    }
+
+    this.withCmd(buildCmdsFromArgs(this.args))
+      .withExposedPorts(
+        ...(this.hasExposedPorts ? this.ports : [CLIENT_PORT, ROUTING_PORT_FOR_CLUSTERING, HTTP_MANAGEMENT_PORT])
+      )
+      .withWaitStrategy(new LogWaitStrategy(new RegExp(".*Server is ready.*")))
```
Comment (cristianrgreco) [inline src/modules/nats/nats-container.ts:68]: We use the static `Wait.forLogMessage('/.../')` in other places

File: package.json [hunk 4]
```diff
@@ -41,6 +41,7 @@
     "docker-compose": "^0.23.17",
     "dockerode": "^3.3.1",
     "get-port": "^5.1.1",
+    "nats": "^2.7.1",
```
Comment (cristianrgreco) [inline package.json:44]: This will need to be a dev dependency

**Comment by cristianrgreco** [comment]
Thanks for the contribution @gmamaladze, left a few comments 🙂

File: src/modules/nats/nats-container.test.ts (hunk 1 above)
Comment (gmamaladze) [reply src/modules/nats/nats-container.test.ts:31]: can be removed

File: package.json (hunk 4 above)
Comment (gmamaladze) [reply package.json:44]: I see.

File: src/modules/nats/nats-container.ts [hunk 5]
```diff
@@ -93,7 +92,42 @@ export class StartedNatsContainer extends AbstractStartedContainer {
     };
   }
 
-  public getConnectionOptions(): ConnectionOptions {
+  public getConnectionOptions(): NatsConnectionOptions {
     return this.connectionOptions;
   }
 }
+
+export interface NatsConnectionOptions {
```
Comment (cristianrgreco) [inline src/modules/nats/nats-container.ts:100]: You will need to export these types via the index.ts 
(so for example you can do `import {NatsContainer} from 'testcontainers'`

**Comment by cristianrgreco** [comment]
@gmamaladze you missed this one: https://github.com/testcontainers/testcontainers-node/pull/375#discussion_r903866034, you won't be able to use it without

**Review by cristianrgreco — APPROVED** [review]

**Comment by cristianrgreco** [comment]
@gmamaladze released in [8.11.0](https://github.com/testcontainers/testcontainers-node/releases/tag/v8.11.0), thanks again for the contribution!

**Pull request merged**